aiosmtpd/docs/handlers.rst

```diff
@@ -177,61 +190,69 @@ of the :ref:`SMTP class<smtp_api>`.
   messages to a remote host and port.  The constructor takes the host name and
   port as positional arguments.  This class cannot be used on the command
   line.
 
-* ``Sink`` - this class just consumes and discards messages.  It's essentially
+* :class:`Sink` - this class just consumes and discards messages.  It's essentially
   the "no op" handler.  It can be used on the command line, but accepts no
   positional arguments.
 
-* ``Message`` - this class is a base class (it must be subclassed) which
+* :class:`Message` - this class is a base class (it must be subclassed) which
   converts the message content into a message instance.  The class used to
   create these instances can be passed to the constructor, and defaults to
-  `email.message.Message`_.
+  :class:`email.message.Message`
 
-  This message instance gains a few additional headers (e.g. ``X-Peer``,
-  ``X-MailFrom``, and ``X-RcptTo``).  You can override this behavior by
+  This message instance gains a few additional headers (e.g. :mailheader:`X-Peer`,
+  :mailheader:`X-MailFrom`, and :mailheader:`X-RcptTo`).  You can override this behavior by
   overriding the ``prepare_message()`` method, which takes a session and an
   envelope.  The message instance is then passed to the handler's
   ``handle_message()`` method.  It is this method that must be implemented in
   the subclass.  ``prepare_message()`` and ``handle_message()`` are both
   called *synchronously*.  This handler cannot be used on the command line.
 
-* ``AsyncMessage`` - a subclass of the ``Message`` handler, with the only
+* :class:`AsyncMessage` - a subclass of the ``Message`` handler, with the only
   difference being that ``handle_message()`` is called *asynchronously*.  This
   handler cannot be used on the command line.
 
-* ``Mailbox`` - a subclass of the ``Message`` handler which adds the messages
-  to a Maildir_.  See below for details.
+* :class:`Mailbox` - a subclass of the ``Message`` handler which adds the messages
+  to a :class:`~mailbox.Maildir`.  See below for details.
 
 
 The Mailbox handler
 ===================
 
 A convenient handler is the ``Mailbox`` handler, which stores incoming
-messages into a maildir::
+messages into a maildir.
 
-    >>> import os
-    >>> from aiosmtpd.controller import Controller
-    >>> from aiosmtpd.handlers import Mailbox
+To try it, let's first prepare an :class:`~contextlib.ExitStack` to automatically
+clean up after we finish:
+
+    >>> from contextlib import ExitStack
     >>> from tempfile import TemporaryDirectory
-    >>> # Clean up the temporary directory at the end of this doctest.
+    >>> # Clean up the temporary directory at the end
+    >>> resources = ExitStack()
     >>> tempdir = resources.enter_context(TemporaryDirectory())
 
+Then, prepare the controller:
+
+    >>> import os
+    >>> from aiosmtpd.controller import Controller
+    >>> from aiosmtpd.handlers import Mailbox
+    >>> #
```

Comment:
Without this empty comment, this piece of doctest wants to separate from the next piece, even though they are next to one another. I can't figure out why an empty line does not work here, while it works in `controller.rst` (lines 141-148)